numpy/core/src/multiarray/dtypemeta.c

```diff
@@ -524,6 +525,42 @@ void_common_instance(PyArray_Descr *descr1, PyArray_Descr *descr2)
     return NULL;
 }
 
+static int
+fill_zero_void_with_objects_strided_loop(
+        void *NPY_UNUSED(traverse_context), PyArray_Descr *descr,
+        char *data, npy_intp size, npy_intp stride,
+        NpyAuxData *NPY_UNUSED(auxdata))
+{
+    PyObject *zero = PyLong_FromLong(0);
+    while (size--) {
+        _fillobject(data, zero, descr);
+        data += stride;
+    }
+    Py_DECREF(zero);
+    return 0;
+}
+
+
+static int
+void_get_fill_zero_loop(void *NPY_UNUSED(traverse_context),
+                        PyArray_Descr *descr,
+                        int NPY_UNUSED(aligned),
+                        npy_intp NPY_UNUSED(fixed_stride),
+                        traverse_loop_function **out_loop,
+                        NpyAuxData **NPY_UNUSED(out_auxdata),
+                        NPY_ARRAYMETHOD_FLAGS *flags)
+{
+    *flags = NPY_METH_NO_FLOATINGPOINT_ERRORS;
+    if (PyDataType_REFCHK(descr)) {
+        *flags |= NPY_METH_REQUIRES_PYAPI;
+        *out_loop = &fill_zero_void_with_objects_strided_loop;
+    }
+    else {
+        *out_loop = NULL;
+    }
+    return 0;
+}
```

Comment:
We should move these to `dtype_traversal.c`, also because that shows how to go about implementing a proper void version?  I had put the clear function setup in `dtype_traversal.c`.  That was the wrong choice I think, having it here is better probably.  (Because I came from the angle of making it similar to casts.)

Reply:
(Just as a note, these two functions really need a proper traversal implementation.)

Reply:
Sorry, a little confused by this comment. Should I move these functions to `dtype_traversal.c` to go with where the clear setup is right now? Or should I move the dtype-specific clear functions here? I'd slightly prefer the latter just so all the dtype-specific functionality for the builtin dtypes are in one file.

Reply:
Wait, I think I put the comment on the wrong line, I wanted to put this on the `dtypemeta.c` loop and get_loop implementations, sorry.  Does it make sense then?

Reply:
I’m confused about what you want me to do, because your comment starts with a suggestion to move the functions to the other file, but then says it was a mistake to put the clear functions there and they should probably live in this file. 

I agree with that and I think it probably makes sense to move the clear get_loop and loop implementations to this file and leave the zero-filling ones I added in this PR here.

Reply:
Arrrrrg!  Attaching the clear functions to the dtype is currently in `dtype_transfer.c` (grouped with the cast setups).  That is weird, and I would agree with moving it into `dtypemeta.c`.

The actual clear traversal loops and get_loops are currently in `dtype_traversal.c` and I think I prefer that because it is quite a lot of code for `dtypemeta.c` and all traversal loop and get_loop follow a very similar pattern.

Reply:
Thanks for clarifying!